pkg/syncer/syncer.go

```diff
@@ -352,10 +369,47 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 				return informers, notSynced
 			},
 		},
-		map[string]controllermanager.ManagedController{},
+		upstreamUpSyncerManagedControllers,
 	)
-	go upstreamUpsyncerControllerManager.Start(ctx)
+	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerStorage) {
```

Comment:
let's start the upstreamUpsyncerControllerManager in any case, and not condition that to the feature gate.